azure-quantum/tests/unit/test_plugins.py

```diff
@@ -0,0 +1,294 @@
+        circuit.cx(1, 2)
+        circuit.measure([0,1,2], [0, 1, 2])
+
+        return circuit
+
+    def test_plugins_submit_qiskit_to_ionq(self):
+        circuit = self._3_qubit_ghz()
+        self._test_qiskit_submit_ionq(circuit=circuit, num_shots=None, num_shots_actual=500)
+    
+    def test_plugins_submit_qiskit_qobj_to_ionq(self):
+        from qiskit import assemble
+        circuit = self._3_qubit_ghz()
+        qobj = assemble(circuit)
+        self._test_qiskit_submit_ionq(circuit=qobj, num_shots=None, num_shots_actual=1024)
+
+    def _test_qiskit_submit_ionq(self, circuit, num_shots, num_shots_actual):
+
+        with unittest.mock.patch.object(
+            Job,
+            self.mock_create_job_id_name,
+            return_value=self.get_test_job_id(),
+        ):
+            workspace = self.create_workspace()
+            provider = AzureQuantumProvider(workspace=workspace)
+            backend = provider.get_backend("ionq.simulator")
+            
+            qiskit_job = backend.run(
+                circuit=circuit,
+                shots=num_shots
+            )
+
+            # Make sure the job is completed before fetching the results
+            # playback currently does not work for repeated calls
+            # See: https://github.com/microsoft/qdk-python/issues/118
+            if not self.is_playback:
+                job = qiskit_job._azure_job
+                self.assertEqual(False, job.has_completed())
+                if self.in_recording:
+                    import time
+                    time.sleep(3)
+                job.refresh()
+                job.wait_until_completed()
+                self.assertEqual(True, job.has_completed())
+                job = provider.get_job(job.id)._azure_job
+                self.assertEqual(True, job.has_completed())
+
+            result = qiskit_job.result()
+            assert result.data()["counts"] == {
+                '000': num_shots_actual//2, '111': num_shots_actual//2
+            }
+            assert result.data()["probabilities"] == {'000': 0.5, '111': 0.5}
+            counts = result.get_counts()
+            assert counts == result.data()["counts"]
+    
+    def test_plugins_retrieve_job(self):
+        with unittest.mock.patch.object(
+            Job,
+            self.mock_create_job_id_name,
+            return_value=self.get_test_job_id(),
+        ):
+            workspace = self.create_workspace()
+            provider = AzureQuantumProvider(workspace=workspace)
+            backend = provider.get_backend("ionq.simulator")
+            circuit = self._3_qubit_ghz()
+            qiskit_job = backend.run(
+                circuit=circuit,
+                num_shots=100
+            )
+            job = qiskit_job._azure_job
+
+            # Make sure the job is completed before fetching the job
+            if not self.is_playback:
+                self.assertEqual(False, job.has_completed())
+                if self.in_recording:
```

Comment:
same comments as above:
- should be using qiskit_job and not the azure job's methods
- should not expect job will actually complete
- should add a timeout.
more over, for this test, is not even necessary wait for job to complete as all we're doing is making sure we can retrieve an existing job.

Reply:
I think we do need to wait for it to complete, we are looking if we can get an existing job that has completed. We could just submit the job and retrieve it but then we can't check if `job.has_completed()` is the same for both jobs as it could have changed between calls.